### PR TITLE
WV-2772: Modernizing date-selector Component

### PR DESCRIPTION
**Pull request by christof-wittreich**
## Description
This converts the date-selector component from a React Class component to a function component, per best modern practices for React. This process included replacing the lifecycle hooks and class state with `useEffect` and `useState`, respectively.

## How To Test
1. `git checkout wv-2772-modernize-dateselector-component`
2. `npm ci`
3. `npm run watch`
4. Locate the date-selector component, which is used as part of the Timeline component in the lower-left.
5. Verify that the new date-selector component functions the same as the existing date-selector component.

### CODE REVIEW

**Review by ryanweiler92 — APPROVED** [review]

File: web/js/components/date-selector/date-selector.js [hunk 1]
```diff
@@ -206,51 +207,47 @@ class DateSelector extends Component {
...
 
     // check if date is within min/max range
-    const dateTime = date.getTime();
+    const dateTime = newDate.getTime();
     const minDateTime = minDate.getTime();
     const maxDateTime = maxDate.getTime();
     const dateWithinRange = dateTime >= minDateTime && dateTime <= maxDateTime;
 
     // updateDate at this stage can still be invalid with pending timeunit changes
     // eslint-disable-next-line react/destructuring-assignment
```
Comment (Tomcariello) [inline web/js/components/date-selector/date-selector.js:233]: I don't think this override is actually doing anything here. Can we remove it?

File: web/js/components/date-selector/date-selector.js [hunk 2]
```diff
@@ -259,19 +256,18 @@ class DateSelector extends Component {
   /**
   * @desc update date with newDate if valid from check and then reset temp time values
   *
-  * @param {String} date
+  * @param {String} dateIn
   * @param {Boolean} isRollDate
   * @returns {void}
   */
   // eslint-disable-next-line react/destructuring-assignment
```
Comment (Tomcariello) [inline web/js/components/date-selector/date-selector.js:263]: I don't think this override is actually doing anything here. Can we remove it?

File: web/js/components/date-selector/date-selector.js [hunk 3]
```diff
@@ -1,61 +1,68 @@
 /* eslint-disable react/jsx-props-no-spreading */
```
Comment (Tomcariello) [inline web/js/components/date-selector/date-selector.js:1]: This override is functional, but it's worth considering getting rid of it (wv-2579 intends to address overrides). It appears we just need to eliminate `{...sharedProps}` when passing props in a handful of places.

**Review by Tomcariello — APPROVED** [review]
This looks good. Feel free to address the overrides (or not) as you see fit.

**Pull request merged**